packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx

```diff
@@ -121,52 +121,23 @@ export const useAuthenticator = (selector?: Selector) => {
   };
 
   /**
-   * For `useSelector`'s selector argument, we just return back the `state`.
-   * The reason is that whenever you select a specific value of the state, the
-   * hook will return *only* that selected value instead of the whole `state`.
+   * For `useSelector`'s selector argument, we transform `state` into
+   * public facade values using `getFacade`.
    *
-   * To provide a consistent set of facade, we let the `selector` trivially return
-   * itself and let comparator decide when to re-render.
+   * This is to hide the internal xstate implementation details to customers.
    */
-  const xstateSelector = (state: AuthMachineState) => state;
-
-  /**
-   * Holds a snapshot copy of last previous facade values. Will be used
-   * on state changes to see if any of facade values have changed.
-   */
-  const prevFacadeRef = React.useRef<ReturnType<typeof getFacade>>();
+  const xstateSelector = (state: AuthMachineState) => getFacade(state);
```

Comment:
Instead of using the hacky `selector` that returns itself, we make the `selector` more explicit by having it return ("select") facade values inferred from current state.

This aligns with the intended purpose of `useSelector`, and works well with `useSyncExternalStore` hook under the hood as well.

Reply:
This is where the compatibility problem occurred -- `useSyncExternalStore` was thinking `state` was not changing because the previous `selector` was just returning the same state object ref.

Reply:
Very nice! Good find, totally makes sense why we weren't getting errors then.